src/Illuminate/Container/BoundMethod.php

```diff
@@ -5,71 +5,78 @@
 use Closure;
 use ReflectionMethod;
 use ReflectionFunction;
+use Illuminate\Support\Arr;
 use InvalidArgumentException;
 
 class BoundMethod
 {
+    /**
+     * @var \Illuminate\Contracts\Container\Container
+     */
+    private static $container;
```

Comment:
What is the need here for memoization? This PR should be focused on fixing the issue at hand, memoization is not needed for that. You had to change some additional method signatures to accomplish that which means more breaking changes than what is needed to fix the problem that this PR is trying to solve. The memoization should be a separate PR. And what are the performance gains here (Taylor usually doesn't merge PRs which supposedly improve the performance of some part of the framework without some benchmarks)?